splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/base/SpliceController.java

```diff
@@ -185,18 +187,19 @@ public boolean batchFetch(List<RowLocation> locations, List<ExecRow> destRows,Fo
             while (results.hasNext()) {
                 DataResult result = results.next();
                 DataValueDescriptor[] destRow = destRows.get(i).getRowArray();
-                if (serializers ==null) {
+                if (serializers == null) {
                     serializers = VersionedSerializers.forVersion(tableVersion, true).getSerializers(destRow);
-                    rowDecoder=new EntryDataDecoder(cols,null,serializers);
+                    rowDecoder = new EntryDataDecoder(cols,null, serializers);
                 }
-                ExecRow row=new ValueRow(destRow.length);
+                ExecRow row = new ValueRow(destRow.length);
                 row.setRowArray(destRow);
                 row.resetRowArray();
-                DataCell keyValue=result.userData();
-                rowDecoder.set(keyValue.valueArray(),keyValue.valueOffset(),keyValue.valueLength());
+                DataCell keyValue = result.userData();
+                rowDecoder.set(keyValue.valueArray(), keyValue.valueOffset(), keyValue.valueLength());
                 rowDecoder.decode(row);
                 i++;
             }
+            assert i == locations.size();
```

Comment:
I believe assertions only get thrown on standalone.  If you want the error to be caught on a cluster as well, maybe an Exception would be better (I don't know this for sure about assertions, but that's the behavior I recall seeing.  You may want to test it).

Reply:
Assertion is not enabled when splice is deployed on a cluster

Reply:
This is development specific and should not happen on a production machine.